opentelemetry-sdk/src/metrics/meter_provider.rs

```diff
@@ -232,10 +232,16 @@ impl MeterProviderBuilder {
     /// with all [Meter]s the [MeterProvider] will create.
     ///
     /// By default, if this option is not used, the default [Resource] will be used.
+    /// Note that the calls to this method are additive, each call merges the provided
+    /// resource with the previous one.
     ///
     /// [Meter]: opentelemetry::metrics::Meter
     pub fn with_resource(mut self, resource: Resource) -> Self {
-        self.resource = Some(resource);
+        self.resource = match self.resource {
+            Some(existing) => Some(existing.merge(&resource)),
+            None => Some(resource),
```

Comment:
Not sure if it is only me, but this is causing confusion. A Resource object represents a single instance with multiple attributes, so `with_resource()` should ideally set the resource rather than merging attributes. If we want to support additive or cooperative resource setting, we could introduce a separate method like `with_merge_resource` or `with_append_resource` (or similiar) to make the merging behavior explicit. This differs from `with_processor`, where multiple processors can be configured.

Reply:
In the end, this is solving the same problem as in:
https://github.com/open-telemetry/opentelemetry-dotnet/issues/2909

Do we want to deviate from a solved problem?

Personally, I was surprised that `with_resource` had replace behavior, but it's purely subjective opinion.

Reply:
OTel .NET was different, as it named its API "Set-Resource", so it was easy to tell that it follows "Set" semantics, and it overrides any existing. It introduced a *new* API called `ConfigureResource`, which operates on the `ResourceBuilder`, and *not* Resource.
In other words, what is proposed in this PR is not same as what .NET had/has.

Reply:
https://github.com/open-telemetry/opentelemetry-rust/issues/2324#issuecomment-2666129606 see my comment on the issue. We can discuss there, and once agreed on the API share, we can come back to this PR.